packages/emitter-framework/src/python/components/class-declaration/class-method.test.tsx

```diff
@@ -0,0 +1,230 @@
+import { Tester } from "#test/test-host.js";
+import { getProgram } from "#test/utils.js";
+import { t } from "@typespec/compiler/testing";
+import { describe, expect, it } from "vitest";
+import { ClassDeclaration } from "../../../../src/python/components/class-declaration/class-declaration.js";
+import { Method } from "../../../../src/python/components/class-declaration/class-method.js";
+import { getOutput } from "../../test-utils.js";
+
+describe("interface methods with a `type` prop", () => {
+  it("creates a class method from an interface method", async () => {
+    const { program, getName } = await Tester.compile(t.code`
+      @test op ${t.op("getName")}(id: string): string;
+    `);
+
+    expect(
+      getOutput(program, [
+        <ClassDeclaration name="basicInterface">
+          <Method async type={getName} />
+        </ClassDeclaration>,
+      ]),
+    ).toRenderTo(`
+      class BasicInterface:
+        async def get_name(self, id: str) -> str:
+          pass
+
+
+    `);
+  });
+
+  it("creates a class method that is a classmethod", async () => {
+    const { program, getName } = await Tester.compile(t.code`
+      @test op ${t.op("getName")}(id: string): string;
+    `);
+
+    expect(
+      getOutput(program, [
+        <ClassDeclaration name="basicInterface">
+          <Method async type={getName} methodType="class" />
+        </ClassDeclaration>,
+      ]),
+    ).toRenderTo(`
+      class BasicInterface:
+        @classmethod
+        async def get_name(cls, id: str) -> str:
```

Comment:
I'm curious about the use of `classmethod` here - as this does not appear to be an idiomatic application of the feature. Normally the `cls` parameter is used to return an instance of that class (i.e. it's most frequently used as an alternative constructor that may require additional processing logic).

Reply:
I agree it's odd, but does that mean it shouldn't be possible?

Reply:
Well that might depend on how common the declaration in TSP is (of which I'm no expert). If this seems like a "natural" or common thing to write in TSP - but it results in unnatural/uncommon Python pattern, then we should probably consider alternatives.
But if this is awkward tsp, and it results in awkward Python - then that's fair, and it probably shouldn't be disallowed on that basis.